src/mscorlib/src/System/Collections/Generic/Dictionary.cs

```diff
@@ -421,13 +444,20 @@ private void Insert(TKey key, TValue value, bool add)
             {
                 if (entries[i].hashCode == hashCode && comparer.Equals(entries[i].key, key))
                 {
-                    if (add)
+                    switch (behavior)
```

Comment:
@mikedn Sure, thanks for the advice. I modified your example to check for `OverwriteExisting` first because that will be the most common branch.